astropy/modeling/physical_models.py

```diff
@@ -371,3 +372,353 @@ def input_units(self):
+        """
+        # Set mass overdensity type and factor
+        if isinstance(massfactor, tuple):
+            # Tuple options
+            #   ("virial")       : virial radius
+            #   ("critical", N)  : radius where density is N that of the critical density
+            #   ("mean", N)      : radius where density is N that of the mean density
+            if massfactor[0].lower() == "virial":
+                # Virial Mass
+                delta = None
+                masstype = massfactor[0].lower()
+            elif massfactor[0].lower() == "critical":
+                # Critical or Mean Overdensity Mass
+                delta = float(massfactor[1])
+                masstype = 'c'
+            elif massfactor[0].lower() == "mean":
+                # Critical or Mean Overdensity Mass
+                delta = float(massfactor[1])
+                masstype = 'm'
+            else:
+                raise ValueError("Massfactor '" + str(massfactor[0]) + "' not one of 'critical', "
+                                                                       "'mean', or 'virial'")
+        else:
+            try:
+                # String options
+                #   virial : virial radius
+                #   Nc  : radius where density is N that of the critical density
+                #   Nm  : radius where density is N that of the mean density
+                if massfactor.lower() == "virial":
+                    # Virial Mass
+                    delta = None
+                    masstype = massfactor.lower()
+                elif massfactor[-1].lower() == 'c' or massfactor[-1].lower() == 'm':
+                    # Critical or Mean Overdensity Mass
+                    delta = float(massfactor[0:-1])
+                    masstype = massfactor[-1].lower()
+                else:
+                    raise ValueError("Massfactor " + str(massfactor) + " string not of the form "
+                                                                       "'#m', '#c', or 'virial'")
+            except (AttributeError, TypeError):
+                raise TypeError("Massfactor " + str(
+                    massfactor) + " not a tuple or string")
+
+        # Set density from masstype specification
+        if masstype == "virial":
+            Om_c = cosmo.Om(redshift) - 1.0
+            d_c = 18.0 * np.pi ** 2 + 82.0 * Om_c - 39.0 * Om_c ** 2
+            self.density_delta = d_c * cosmo.critical_density(redshift)
+        elif masstype == 'c':
+            self.density_delta = delta * cosmo.critical_density(redshift)
+        elif masstype == 'm':
+            self.density_delta = delta * cosmo.critical_density(redshift) * cosmo.Om(redshift)
+        else:
+            raise ValueError("Invalid masstype '" + str(masstype) +
+                             "'. Should be one of 'virial','c', or 'm'")
+        return self.density_delta
+
+    @staticmethod
+    def A_NFW(y):
+        r"""
+        Commonly used expression in several calculations.
```

Comment:
```suggestion
        Dimensionless volume integral of the NFW profile, used as an intermediate step in some calculations for this model.
```

Reply:
I really like this comment. Change made.